corehq/apps/userreports/pillow.py

```diff
@@ -201,15 +202,22 @@ def migrate_tables(self, engine, diffs, table_names, adapters_by_table):
             adapter = adapters_by_table[table]
             adapter.log_table_migrate(source='pillowtop', diffs=diffs)
 
-    def rebuild_table(self, adapter):
+    def rebuild_table(self, adapter, diffs=None):
         config = adapter.config
         if not config.is_static:
             latest_rev = config.get_db().get_rev(config._id)
             if config._rev != latest_rev:
                 raise StaleRebuildError('Tried to rebuild a stale table ({})! Ignoring...'.format(config))
-            adapter.rebuild_table(source='pillowtop')
-        else:
+
+        if config.disable_destructive_rebuild and adapter.table_exists:
+            diff_dicts = [diff.to_dict() for diff in diffs]
+            adapter.log_table_rebuild_skipped(source='pillowtop', diffs=diff_dicts)
+            return
+
+        if config.is_static:
             rebuild_indicators.delay(adapter.config.get_id, source='pillowtop')
+        else:
+            adapter.rebuild_table(source='pillowtop')
```

Comment:
why this alternate method

Reply:
It looks like its keeping the previous behavior. I'm guessing that the original intention was that rebuilds (with data) should happen automatically for static data sources. For non-static, report builder or the user should manually trigger rebuilds.

probably deserves some rethinking though not in this PR

Reply:
ah yep, i missed the deletion in line 210